src/Illuminate/Foundation/Console/MigrateAutoCommand.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace Illuminate\Foundation\Console;
+
+use Doctrine\DBAL\Schema\Comparator;
+use Illuminate\Console\Command;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Filesystem\Filesystem;
+use Illuminate\Support\Facades\Artisan;
+use Illuminate\Support\Facades\Schema;
+
+class MigrateAutoCommand extends Command
+{
+    protected $signature = 'migrate:auto {--fresh} {--seed} {--force}';
+
+    public function handle()
+    {
+        Artisan::call('migrate' . ($this->option('fresh') ? ':fresh' : null) . ($this->option('force') ? ' --force' : null));
+
+        $filesystem = new Filesystem;
+        $dir = base_path('app/Models');
+
+        if ($filesystem->exists($dir)) {
+            foreach ($filesystem->allFiles($dir) as $file) {
+                $class = app('App\\Models\\' . str_replace(['/', '.php'], ['\\', ''], $file->getRelativePathname()));
+
+                if (method_exists($class, 'migration')) {
```

Comment:
OMG, model knows about factories, now they will know about migrations? What about versioning, if model would be change? What about migration's drop?

Reply:
Doctrine handles all of that via the `diffTable` method.  https://github.com/doctrine/dbal/blob/2.12.x/lib/Doctrine/DBAL/Schema/Comparator.php

Reply:
How you plan to drop the initial migration? What if on project we use a module architecture? `model_path` option in config  would be useless. You must add an option in command that ask to choose models path, if you want, or use default.

Reply:
You can still use this alongside traditional migrations for more complex applications. It runs the traditional migrations before the automatic migration methods. We could also add a `--path` option to the command as well, in case you want to specify the model path for the command to use in edge cases.